src/app/zap-templates/templates/app/helper.js

```diff
@@ -348,7 +348,7 @@ function asMEI(prefix, suffix)
   return cHelper.asHex((prefix << 16) + suffix, 8);
 }
 
-function asChipZapType(type)
+function asChipZapType(type, makeDecodable)
```

Comment:
Wait, why does this function exist?  How does it differ from `atomicType` in `src/app/zap-templates/common/override.js`?  Is there a reason it does not include all the types `atomicType` does?  @vivien-apple @yunhanw-google 

At the very least all of this needs to be documented on the two functions, so people know which one to use where.  But in general, it would be best if we had one source of truth for this sort of mapping, not multiple out-if-sync copies of it.  And even better if the function names were clear enough so you could tell from the function name what it does....  I think `atomicType` is named that because the ZAP core expects a function with that name, but we should really have a function with a sane name (`zapTypeToMatterType`?  Is that what it takes as input and produces as output?) that `atomicType` could call and templates coud use directly...

Probably out of scope for this PR, especially because this function and atomicType don't even agree on what types things map to(!), but sorting this out should be a top priority.  Because again, this thing does not seem to return all the types we might need (e.g. I don't see it returning `chip::GroupId` or `chip::NodeId` or anything like that, which seems very suspicious to me).

What _should_ happen in this PR is that we should have functions named `zapTypeToDecodableType` and `zapTypeToEncodableType` instead of an inscrutable boolean arg in the templates.  The implementation can use a helper with a nicely documented boolean arg, of course.